modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/ManagementToolbarTag.java

```diff
@@ -844,7 +887,71 @@ protected int processStartTag() throws Exception {
 			jspWriter.write("</span></button></div></li>");
 		}
 
-		if (getSortingURL() != null) {
+		List<DropdownItem> orderDropdownItems = getOrderDropdownItems();
+
+		if (FFManagementToolbarConfigurationUtil.showDesignImprovements() &&
+			!active && (orderDropdownItems != null) &&
+			(orderDropdownItems.size() > 1)) {
+
+			jspWriter.write("<li class=\"nav-item\"><div class=\"dropdown\">");
+			jspWriter.write("<button class=\"btn btn-unstyled dropdown-toggle");
+			jspWriter.write(" ml-2 mr-2 nav-link\"");
+
+			if (disabled) {
+				jspWriter.write(" disabled");
+			}
+
+			jspWriter.write(" type=\"button\"><span class=\"");
+			jspWriter.write("navbar-breakpoint-down-d-none\"><span class=\"");
+			jspWriter.write("inline-item inline-item-before\">");
+
+			iconTag = new IconTag();
+
+			String orderSymbol = "order-list-down";
+
+			if (getSortingOrder().equals("asc")) {
+				orderSymbol = "order-list-up";
+			}
+
+			iconTag.setSymbol(orderSymbol);
+
+			iconTag.doTag(pageContext);
+
+			jspWriter.write("</span>");
+
+			jspWriter.write("<span class=\"navbar-text-truncate\">");
+			jspWriter.write(LanguageUtil.get(resourceBundle, "order"));
+			jspWriter.write("</span>");
+
+			iconTag = new IconTag();
+
+			iconTag.setCssClass("inline-item inline-item-after");
+			iconTag.setSymbol("caret-bottom");
+
+			iconTag.doTag(pageContext);
+
+			jspWriter.write("</span><span class=\"navbar-breakpoint-d-none\">");
+
+			iconTag = new IconTag();
+
+			iconTag.setSymbol(orderSymbol);
+
+			iconTag.doTag(pageContext);
+
+			jspWriter.write("</span></button></div></li>");
+		}
+
+		Boolean hasOneOrderDropdownItem =
```

Comment:
We can squash `hasOneOrderDropdownItem` and `hasNoOrderDropdownItems` into something like `showOrderDropdown` since we're only using both of them in the conditional